framework/test/unit/modules/validators/api.spec.ts

```diff
@@ -201,158 +211,334 @@ describe('ValidatorsModuleAPI', () => {
+			);
+
+			let genWithCountGreaterThanOne = 0;
+			for (const generatorAddress of Object.keys(result)) {
+				if ((result[generatorAddress] as number) > 1) {
+					genWithCountGreaterThanOne += 1;
+				}
+			}
+
+			let genWithCountGreaterThanTwo = 0;
+			for (const generatorAddress of Object.keys(result)) {
+				if ((result[generatorAddress] as number) > 2) {
+					genWithCountGreaterThanTwo += 1;
+				}
+			}
+
+			expect(generatorList).toHaveLength(genWithCountGreaterThanOne);
+			expect(genWithCountGreaterThanTwo).toBeGreaterThan(0);
+		});
+
+		it('should be able to return atleast one generator if input timestamps are valid and difference between input timestamps is zero', async () => {
+			await genesisDataSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			const result = await validatorsModule.api.getGeneratorsBetweenTimestamps(
+				apiContext,
+				genesisTimestamp,
+				genesisTimestamp,
+			);
+
+			expect(Object.keys(result).length).toBeGreaterThan(0);
+		});
+
+		it('should be able to return atleast one generator if input timestamps are valid and difference between input timestamps is less than block time ', async () => {
+			const blockTime = 10;
+
+			await genesisDataSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			const result = await validatorsModule.api.getGeneratorsBetweenTimestamps(
+				apiContext,
+				genesisTimestamp,
+				genesisTimestamp + blockTime - 1,
+			);
+
+			expect(Object.keys(result).length).toBeGreaterThan(0);
 		});
 
-		it(`should throw if input timestamps are invalid`, async () => {
+		it('should be able to return more than one generators if input timestamps are valid and difference between input timestamps is equal to block time ', async () => {
+			const blockTime = 10;
+
+			await genesisDataSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				EMPTY_KEY,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			const result = await validatorsModule.api.getGeneratorsBetweenTimestamps(
+				apiContext,
+				genesisTimestamp,
+				genesisTimestamp + blockTime,
+			);
+
+			expect(Object.keys(result).length).toBeGreaterThan(1);
```

Comment:
```suggestion
			expect(Object.keys(result).length).toBe(2);
```
or use `toHaveLength(2)` in general